projects/ad713x_fmcz/ad713x_fmc.c

```diff
@@ -79,126 +88,170 @@
+	ad713x_init_param_1.clk_delay_en = false;
+	ad713x_init_param_1.crc_header = CRC_8;
+	ad713x_init_param_1.dev_id = ID_AD7134;
+	ad713x_init_param_1.format = QUAD_CH_PO;
+	ad713x_init_param_1.gpio_dclkio.number = GPIO_DCLKIO_1;
+	ad713x_init_param_1.gpio_dclkio.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_1.gpio_dclkmode.number = GPIO_DCLKMODE;
+	ad713x_init_param_1.gpio_dclkmode.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_1.gpio_mode.number = GPIO_MODE_1;
+	ad713x_init_param_1.gpio_mode.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_1.gpio_pnd.number = GPIO_PDN_1;
+	ad713x_init_param_1.gpio_pnd.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_1.gpio_resetn.number = GPIO_RESETN_1;
+	ad713x_init_param_1.gpio_resetn.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_1.mode_master_nslave = true;
+	ad713x_init_param_1.dclkmode_free_ngated = false;
+	ad713x_init_param_1.dclkio_out_nin = false;
+	ad713x_init_param_1.pnd = true;
+	ad713x_init_param_1.spi_init_prm.chip_select = AD7134_1_SPI_CS;
+	ad713x_init_param_1.spi_init_prm.max_speed_hz = 100000000;
+	ad713x_init_param_1.spi_init_prm.mode = SPI_MODE_3;
+	ad713x_init_param_1.spi_init_prm.extra = (void *)&spi_ps_extra;
+	ad713x_init_param_1.spi_common_dev = 0;
+
+	ad713x_init_param_2.adc_data_len = ADC_24_BIT_DATA;
+	ad713x_init_param_2.clk_delay_en = false;
+	ad713x_init_param_2.crc_header = CRC_8;
+	ad713x_init_param_2.dev_id = ID_AD7134;
+	ad713x_init_param_2.format = QUAD_CH_PO;
+	ad713x_init_param_2.gpio_dclkio.number = GPIO_DCLKIO_2;
+	ad713x_init_param_2.gpio_dclkio.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_2.gpio_dclkmode.number = GPIO_DCLKMODE;
+	ad713x_init_param_2.gpio_dclkmode.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_2.gpio_mode.number = GPIO_MODE_2;
+	ad713x_init_param_2.gpio_mode.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_2.gpio_pnd.number = GPIO_PDN_2;
+	ad713x_init_param_2.gpio_pnd.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_2.gpio_resetn.number = GPIO_RESETN_2;
+	ad713x_init_param_2.gpio_resetn.extra = (void *)&gpio_extra_param;
+	ad713x_init_param_2.mode_master_nslave = false;
+	ad713x_init_param_2.dclkmode_free_ngated = false;
+	ad713x_init_param_2.dclkio_out_nin = false;
+	ad713x_init_param_2.pnd = true;
+	ad713x_init_param_2.spi_init_prm.chip_select = AD7134_2_SPI_CS;
+	ad713x_init_param_2.spi_init_prm.max_speed_hz = 100000000;
+	ad713x_init_param_2.spi_init_prm.mode = SPI_MODE_3;
+	ad713x_init_param_2.spi_init_prm.extra = (void *)&spi_ps_extra;
+	ad713x_init_param_2.spi_common_dev = 0;
 
-	spi_eng_msg_cmds[0] = SLEEP(2000); // 2us
-	spi_eng_msg_cmds[1] = TRANSFER_BYTES_R(3);
+	spi_eng_msg_cmds[0] = SLEEP(100);
+	spi_eng_msg_cmds[1] = TRANSFER_BYTES_R(1);
 
 	Xil_ICacheEnable();
 	Xil_DCacheEnable();
 
-	ad713x_init(&dev, ad713x_default_init_param);
+	ad713x_init(&ad713x_dev_1, &ad713x_init_param_1);
 
-	spi_eng_init(&desc, spi_eng_init_params);
+	mdelay(1000);
```

Comment:
If delay is needed after init probably it should be placed inside `ad713x_init` function

Reply:
Unfortunately, I am not sure the delay is needed. Per datasheet is not, but I have not tested without it so I would have to do that when I recover the board.